### PR TITLE
Check and update model list in index.rst automatically

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

We currently have two lists of the models to maintain (in the main README and in the `index.rst` for the docs) which is painful. This script checks the mode list in the index.rst is a proper converted copy of the model list in the README and can also fix it with the command `make fix-copies` (same API as for the copies of parts of the model). It also enforces our maximum character per line so that the `index.rst` is still readable in an editor.

### CODE REVIEW

File: utils/check_copies.py [hunk 1]
```diff
@@ -166,6 +167,113 @@ def check_copies(overwrite: bool = False):
...
+        else:
+            current_line = f"{current_line} {word}"
+    lines.append(current_line)
+    return "\n".join(lines)
+
+
+def convert_to_rst(model_list, max_per_line=None):
+    """ Convert `model_list` to rst format. """
+    # Convert **[description](link)** to `description <link>`__
+    model_list = re.sub(r"\*\*\[([^\]]*)\]\(([^\)]*)\)\*\*", r"`\1 <\2>`__", model_list)
```
Comment (sgugger) [inline utils/check_copies.py:222]: That feeling when this regex works on the first try ;-)

Comment (LysandreJik) [reply utils/check_copies.py:222]: Very nice!

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Comment by LysandreJik** [comment]
I'll let you rebase!

**Pull request merged**